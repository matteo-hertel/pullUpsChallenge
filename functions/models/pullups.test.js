const { pullupModel, makePullup } = require("./pullups.js");
const getDiffentKeys = keys => {
  return function(key) {
    return keys.indexOf(key) === -1 ? true : false;
  };
};
describe("Pullups model", () => {
  it("should not contain extraneous keys", () => {
    const unsanitisedObject = {
      amount: 50,
      foo: "bar",
      bar: "foo"
    };

    const set = makePullup(unsanitisedObject);

    expect(
      Object.keys(set).filter(getDiffentKeys(Object.keys(pullupModel)))
    ).toHaveLength(0);
  });

  it("should be able to update as many time as neded", () => {
    const finalForm = {
      amount: 50,
      completed: true,
      todoistID: "todoistID"
    };

    const set = makePullup({});
    const update1 = makePullup({ amount: finalForm.amount }, set);
    const update2 = makePullup({ completed: finalForm.completed }, update1);

    const finalUpdate = makePullup({ todoistID: finalForm.todoistID }, update2);
    Object.keys(finalForm).map(k =>
      expect(finalUpdate[k]).toEqual(finalForm[k])
    );
  });
});
