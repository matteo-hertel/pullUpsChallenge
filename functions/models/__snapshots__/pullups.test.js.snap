// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pullups model should match snapshot 1`] = `
Object {
  "amount": 0,
  "cached": false,
  "completed": false,
  "date": 1520208000000,
  "dateString": "2018-03-05",
  "rejected": false,
  "todoistID": "",
  "weekNumber": 10,
}
`;
